inspirehep/modules/orcid/api.py

```diff
@@ -53,50 +53,42 @@
 LOGGER = getStackTraceLogger(__name__)
 
 
-def push_record_with_orcid(recid, orcid, oauth_token, putcode=None, old_hash=None):
+def push_record_with_orcid(recid, orcid, oauth_token):
     """Push record to ORCID with a specific ORCID ID.
 
     Args:
         recid (string): HEP record to push
         orcid (string): ORCID identifier to push onto
         oauth_token (string): ORCID user OAUTH token
-        putcode (Union[string, NoneType]): put-code to push record onto,
-            if None will push as a new record
-        old_hash (Optional[string]): previous hash of the record
 
     Returns:
         Tuple[string, string]: a tuple with two elements:
             - the put-code of the inserted item,
             - and the new hash of the ORCID record
     """
-    record = get_db_record('lit', recid)
-
+    inspire_record = get_db_record('lit', recid)
     orcid_cache = OrcidCache(orcid)
 
-    if not putcode or not old_hash:
-        cached_putcode, cached_old_hash = orcid_cache.read_record_data(recid)
-        putcode = putcode or cached_putcode
-        old_hash = old_hash or cached_old_hash
-
-    new_hash = OrcidHasher().compute_hash(record)
-    if new_hash == old_hash:
-        LOGGER.info(
-            'Hash unchanged: not pushing #%s as not a meaningful update', recid
-        )
-        return putcode, new_hash
+    putcode = orcid_cache.read_work_putcode(recid)
+    if not orcid_cache.has_work_content_changed(recid, inspire_record):
+        LOGGER.info('Hash unchanged: not pushing #%s as not a meaningful update', recid)
```

Comment:
add the orcid to the log?

Reply:
All log statements will be re-written in my next changes when I am tackling log aggregation and monitoring. Anyway, I'll add it.

Reply:
Done